core/templates/services/speech-synthesis-chunker.service.ts

```diff
@@ -175,29 +186,35 @@ export class SpeechSynthesisChunkerService {
     elt.find('oppia-noninteractive-' + this.RTE_COMPONENT_NAMES.Link)
       .replaceWith(function() {
         var element = <HTMLElement> this;
-        if (element.attributes['text-with-value'] !== undefined) {
-          const newTextContent = element.attributes[
-            'text-with-value'].textContent.replace(/&quot;/g, '');
-          // Variable newTextContent ends with a " character, so this is being
-          // ignored in the condition below.
-          return newTextContent && newTextContent !== '"' ?
-            newTextContent + ' ' : '';
-        }
+        const _newTextAttr = element.attributes[
+          <keyof NamedNodeMap> 'text-with-value'] as Attr;
+        // 'Node.textContent' only returns 'null' if the Node is a
+        // 'document' or a 'DocType'. '_newTextAttr' is neither.
+        const newTextContent = _newTextAttr.textContent?.replace(
+          /&quot;/g, '');
+        // Variable newTextContent ends with a " character, so this is being
+        // ignored in the condition below.
+        return (
+          newTextContent && newTextContent !== '"' ?
+          newTextContent + ' ' : ''
+        );
       });
 
     var _this = this;
     // Convert LaTeX to speakable text.
     elt.find('oppia-noninteractive-' + this.RTE_COMPONENT_NAMES.Math)
       .replaceWith(function() {
         var element = <HTMLElement> this;
-        if (element.attributes['math_content-with-value'] !== undefined) {
-          var mathContent = <MathExpressionContent>(
-            _this.htmlEscaper.escapedJsonToObj(
-              element.attributes['math_content-with-value'].textContent));
-          const latexSpeakableText = _this._formatLatexToSpeakableText(
-            mathContent.raw_latex);
-          return latexSpeakableText.length > 0 ? latexSpeakableText + ' ' : '';
-        }
+        const _mathContentAttr = element.attributes[
+          <keyof NamedNodeMap> 'math_content-with-value'] as Attr;
+        var mathContent = (
+          <MathExpressionContent>(_this.htmlEscaper.escapedJsonToObj(
+            // 'Node.textContent' only returns 'null' if the Node is a
+            // 'document' or a 'DocType'. '_mathContentAttr' is neither.
+            <string> _mathContentAttr.textContent)));
+        const latexSpeakableText = _this._formatLatexToSpeakableText(
+          mathContent.raw_latex);
+        return latexSpeakableText.length > 0 ? latexSpeakableText + ' ' : '';
```

Comment:
Considering the logic is changed (somewhat) here, have you tested this part manually?

Reply:
@DubeySandeep yes, have tested this